sympy/functions/elementary/integers.py

```diff
@@ -587,6 +573,42 @@ def _value_one_or_more(self, other):
             if other.is_integer and other.is_positive:
                 return S.true
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if r.is_zero:
+                ndir = arg.dir(x, cdir=cdir)
+                return r + 1 if ndir.is_negative else r
```

Comment:
```
In [1]: frac(x)._eval_as_leading_term(x)                                        
Out[1]: 0  # x expected
```

Reply:
I've fixed this now !

Reply:
Some examples being
```
>>> frac(sin(x)).as_leading_term(x, cdir=1)
sin(x)
>>> frac(sin(x)).as_leading_term(x, cdir=-1)
1
>>> frac(sin(x**2)).as_leading_term(x, cdir=1)
sin(x**2)
>>> frac(sin(x**2)).as_leading_term(x, cdir=-1)
sin(x**2)
```
The examples I tried out looks fine to me !